Error with the descriptor resources when using fast-jar

**Describe the bug**
```
Caused by: java.io.IOException: Failed to locate codestarts/quarkus on the classpath
        at io.quarkus.platform.descriptor.loader.json.ClassPathResourceLoader.loadResourceAsPath(ClassPathResourceLoader.java:26)
        at io.quarkus.platform.descriptor.loader.json.impl.QuarkusJsonPlatformDescriptor.loadResourceAsPath(QuarkusJsonPlatformDescriptor.java:144)
        at io.quarkus.devtools.codestarts.QuarkusPlatformCodestartResourceLoader.loadResourceAsPath(QuarkusPlatformCodestartResourceLoader.java:19)
        at io.quarkus.devtools.codestarts.CodestartCatalogLoader.loadCodestarts(CodestartCatalogLoader.java:63)
        at io.quarkus.devtools.codestarts.CodestartCatalogLoader.loadCodestarts(CodestartCatalogLoader.java:51)
        at io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartCatalog.fromQuarkusPlatformDescriptor(QuarkusCodestartCatalog.java:64)
        at io.quarkus.devtools.commands.handlers.CreateProjectCommandHandler.execute(CreateProjectCommandHandler.java:107)

```


It seems something is wrong with the resources names.

--- a/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java
+++ b/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java
@@ -125,15 +125,12 @@ public class SerializedApplication {
                 int numDirs = in.readUnsignedShort();
                 for (int i = 0; i < numDirs; ++i) {
                     String dir = in.readUTF();
-                    ClassLoadingResource[] existing = resourceDirectoryMap.get(dir);
-                    if (existing == null) {
-                        resourceDirectoryMap.put(dir, new ClassLoadingResource[] { resource });
-                    } else {
-                        ClassLoadingResource[] newResources = new ClassLoadingResource[existing.length + 1];
-                        System.arraycopy(existing, 0, newResources, 0, existing.length);
-                        newResources[existing.length] = resource;
-                        resourceDirectoryMap.put(dir, newResources);
+                    int j = dir.indexOf('/');
+                    while (j >= 0) {
+                        addResourceDir(dir.substring(0, j), resource, resourceDirectoryMap);
+                        j = dir.indexOf('/', j + 1);
                     }
+                    addResourceDir(dir, resource, resourceDirectoryMap);
                 }
             }
             int packages = in.readUnsignedShort();
@@ -165,6 +162,19 @@ public class SerializedApplication {
                 classLoadingResource.init(runnerClassLoader);
             }
             return new SerializedApplication(runnerClassLoader, mainClass);
+        }
+    }
+
+    private static void addResourceDir(String dir, JarResource resource,
+            Map<String, ClassLoadingResource[]> resourceDirectoryMap) {
+        ClassLoadingResource[] existing = resourceDirectoryMap.get(dir);
+        if (existing == null) {
+            resourceDirectoryMap.put(dir, new ClassLoadingResource[] { resource });
+        } else {
+            ClassLoadingResource[] newResources = new ClassLoadingResource[existing.length + 1];
+            System.arraycopy(existing, 0, newResources, 0, existing.length);
+            newResources[existing.length] = resource;
+            resourceDirectoryMap.put(dir, newResources);
         }
     }
 

